(FACT-2564) Add support for zpool_featureflags and fix zpool_version

smortex opened the pull request:
This allows facter-ng to return the same facts as facter 3.x.

This changes took place in facter 3.8.0 and was discussed here:
https://github.com/puppetlabs/facter/pull/1597

smortex committed:
(FACT-2564) Add support for zpool_featureflags and fix zpool_version

This allows facter-ng to return the same facts as facter 3.x.

This changes took place in facter 3.8.0:
https://github.com/puppetlabs/facter/pull/1597

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

oanatmaria commented:
Hi, could you please open a Jira ticket for this changes?

oanatmaria commented on lib/resolvers/solaris/zpool_resolver.rb:
```diff
@@ -22,11 +22,13 @@ def zpool_fact(fact_name)
 
             @fact_list[:zpool_featurenumbers] = features_list.join(',')
-            @fact_list[:zpool_version] = features_list.last
+            @fact_list[:zpool_featureflags] = features_flags.join(',')
+            @fact_list[:zpool_version] = features_flags.any? ? 5000 : features_list.last
```
could you write some tests for these changes?

smortex replied:
Done!

smortex commented:
I just opened https://tickets.puppetlabs.com/browse/FACT-2564, added tests as requested and rebased the whole on top of master with a reference to the Jira issue in the squashed commit message. Thanks!

Filipovici-Andrei commented on spec/facter/resolvers/solaris/zpool_resolver_spec.rb:
```diff
@@ -28,6 +28,23 @@
+
+      it 'returns zpool featureflags fact' do
+        result = Facter::Resolvers::Solaris::ZPool.resolve(:zpool_featureflags)
+        expect(result).to eq('async_destroy,empty_bpobj,lz4_compress,multi_vdev_crash_dump,' \
```
Could you move the expected feature flags into a separate let() like in the zpool_featureflags_spec?

smortex replied:
Sure!

smortex committed:
(FACT-2564) Move zpool feature flags in a let statement

Commit:
Merge branch 'master' into zpool_featureflags_and_version
